code/ModelAdmin.php

```diff
@@ -202,6 +203,9 @@ function ($form) {
                 $this->extend('updateSearchForm', $form);
             }
         ));
+        // GridFieldPaginator has to be added after filter header for it to function correctly
+        $listField->getConfig()->removeComponentsByType(GridFieldPaginator::class);
+        $listField->getConfig()->addComponent(new GridFieldPaginator());
```

Comment:
Can you please use Injector to resolve this component?

Reply:
Done